### PR TITLE
Add updating publishing-bot rules to the Branch Management handbook and Release issue template

### PR DESCRIPTION
**Pull request by xmudrii**
#### What type of PR is this:

/kind documentation

#### What this PR does / why we need it:

In the previous (1.20) and the current (1.21) release cycles, we encountered a problem that the publishing-bot rules were not updated when creating a new release branch. That caused a delay between the RC release and creating tags for the RC release in the staging repositories.

The current workflow is that Release Managers cut the RC (which creates a new branch) and then the publishing-bot folks are responsible to add rules for a newly-created branch.

To remind the publishing-bot folks, we create an issue in the `k8s-release-robot/sig-release` repository, such as this one: https://github.com/k8s-release-robot/sig-release/issues/30

However, due to timezones and availability of publishing-bot folks, there might be a delay between the branch cut and when they can update the publishing-bot rules. That delay might even be one day or more. Additionally, once the rules are updated, it might take up to 5 hours for the publishing-bot to publish branches and tags (the bot runs every 4 hours and each run is 1 hour long).

This RFC proposes that the publishing-bot folks handover the responsibility of adding publishing-rules for a new branch to the Release Managers.

I believe that this task is a low to medium effort for the Release Managers, but it can remove or significantly reduce the mentioned delay.

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:

Additional discussion and feedback can be found in the following Slack thread: https://kubernetes.slack.com/archives/C2C40FMNF/p1617007926111900

/hold
for discussion

/cc @kubernetes/release-managers @kubernetes/sig-release-leads

/cc @nikhita @sttts @dims 
tagging folks responsible for the publishing-bot

### CODE REVIEW

**Comment by dims** [comment]
@xmudrii happy for release managers to create the PR needed for sure! i'd leave the `/approve` for folks who currently take care of the bot.

**Comment by saschagrunert** [comment]
Sounds good to me, too.

File: release-engineering/role-handbooks/branch-manager.md
Comment (nikhita) [inline release-engineering/role-handbooks/branch-manager.md:609]: This won't necessarily work every time because dependencies could have been updated between the previous release and the current release.

Instead, we should copy the rules of the `master` branch and then change the branch name.

**Review by nikhita — COMMENTED** [review]
>  i'd leave the /approve for folks who currently take care of the bot.

Agree.

If you could add a note to ping the publishing-bot OWNERS for approval, that'd be great :+1:

**Comment by nikhita** [comment]
Created a help-wanted issue to automate this - https://github.com/kubernetes/publishing-bot/issues/252

File: .github/ISSUE_TEMPLATE/cut-release.md
Comment (puerco) [inline .github/ISSUE_TEMPLATE/cut-release.md:23]: :thinking: I don't know about listing this one here, there are lots of other things related to the branch creation which have to be performed. We should either have a global __:black_square_button: do branch-creation things__ checkbox or nothing at all, wdyt?

**Comment by palnabarun** [comment]
@xmudrii -- can you please have a look at the review comments above?

**Comment by xmudrii** [comment]
@palnabarun -- the PR is intentionally on hold until we don't figure out are we going to proceed with https://github.com/kubernetes/publishing-bot/pull/253. I'll ping the PR author to see is there any update.

**Comment by palnabarun** [comment]
I think we should not block ourselves on kubernetes/publishing-bot#253. We should at least document for branch managers that they would need to update the publishing bot rules. This is an easy to miss step amidst all the steps needed to be performed during an RC cut.

File: .github/ISSUE_TEMPLATE/cut-release.md
Comment (palnabarun) [reply .github/ISSUE_TEMPLATE/cut-release.md:23]: @puerco -- whenever you can, can you please enumerate the other things that need to be performed during branch creation that may not be in the handbook?

**Comment by justaugustus** [comment]
> I think we should not block ourselves on [kubernetes/publishing-bot#253](https://github.com/kubernetes/publishing-bot/pull/253). We should at least document for branch managers that they would need to update the publishing bot rules. This is an easy to miss step amidst all the steps needed to be performed during an RC cut.

Agreed.
@xmudrii -- can your rebase?

**Comment by xmudrii** [comment]
@justaugustus — I’ll take care of it tomorrow.

**Comment by justaugustus** [comment]
> @justaugustus — I’ll take care of it tomorrow.

SGTM. Thanks Marko!

File: .github/ISSUE_TEMPLATE/cut-release.md
Comment (puerco) [reply .github/ISSUE_TEMPLATE/cut-release.md:23]: I think in the handbook the list is mostly complete. I think the last time i cut the branch some instructions for test infra were outdated (I noticed @spiffxp fixed some of the generated jobs, for example). But [the list itself](https://github.com/kubernetes/sig-release/blob/master/release-engineering/role-handbooks/branch-manager.md#branch-creation) seems ok.

**Review by justaugustus — COMMENTED** [review]
@xmudrii -- Looking at this again side-by-side with the publishing-bot repo and https://github.com/kubernetes/kubernetes/tree/master/staging, I think it would maybe be a better idea for https://github.com/kubernetes/publishing-bot to be the canonical location for this content and we link out to it like:

```md
See [k/publishing-bot](https://git.k8s.io/publishing-bot) for instructions on how to do "foo".
```

Thoughts @kubernetes/publishing-bot-reviewers @kubernetes/release-engineering?

**Comment by xmudrii** [comment]
@justaugustus -- Created a PR in k/publishing-bot to document this: https://github.com/kubernetes/publishing-bot/pull/283

**Comment by xmudrii** [comment]
@kubernetes/release-engineering Updated and rebased the PR. PTAL.

**Review by justaugustus — APPROVED** [review]
Thanks Marko!
/hold cancel

**Pull request merged**
